Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Instance/InstanceDomGenerator.cpp

```diff
@@ -200,38 +202,57 @@ namespace AzToolsFramework
             }
         }
 
-        void InstanceDomGenerator::UpdateContainerEntityInDomFromRoot(PrefabDom& instanceDom, const Instance& instance) const
+        void InstanceDomGenerator::UpdateContainerEntityInDomFromHighestAncestor(PrefabDom& instanceDom, const Instance& instance) const
         {
             // TODO: Modifies the function so it updates the transform only.
 
             InstanceClimbUpResult climbUpResult = PrefabInstanceUtils::ClimbUpToTargetOrRootInstance(instance, nullptr);
-            const Instance* rootInstancePtr = climbUpResult.m_reachedInstance;
-            AZStd::string relativePathFromRoot = PrefabInstanceUtils::GetRelativePathFromClimbedInstances(climbUpResult.m_climbedInstances);
 
             // No need to update the instance DOM if the given instance is the root instance.
-            if (rootInstancePtr == &instance)
+            if (climbUpResult.m_reachedInstance == &instance)
             {
                 return;
             }
 
-            // Creates a path from the root instance to container entity of the given insance.
-            const AZStd::string pathToContainerEntity =
-                AZStd::string::format("%s/%s", relativePathFromRoot.c_str(), PrefabDomUtils::ContainerEntityName);
-            PrefabDomPath domPathToContainerEntity(pathToContainerEntity.c_str());
-
-            // Retrieves the container entity DOM of the root template DOM.
+            // Get the instance's container entity DOM from the top-most ancestor that has it.
+            // An ancestor DOM may not have the data if it has deleted the instance as an override
             PrefabDom containerEntityDom;
+
+            AZStd::vector<const Instance*>& climbedInstances = climbUpResult.m_climbedInstances;
+            const Instance* topInstance = climbUpResult.m_reachedInstance;
+            for (auto instanceIter = climbedInstances.crbegin(); instanceIter != climbedInstances.crend(); ++instanceIter)
```

Comment:
Just want to double check I understand it right.
```cpp
const Instance* topInstance = climbUpResult.m_reachedInstance;
```
At this line, topInstance will always be initialized as the root instance?

```cpp
for (auto instanceIter = climbedInstances.crbegin(); instanceIter != climbedInstances.crend(); 
     ++instanceIter) {}
```
It usually just stops in the first instance (i.e. root instance) and if it could not find the DOM it will end up fetching the container entity DOM from itself?

Reply:
Actually after the change to share code, it will und up fetching from itself. However "UpdateContainerEntityInDom" already returns early if reached instance is itself because it doesn't have to do anything in that case.